src/Tables.jl

```diff
@@ -202,6 +202,25 @@ function Base.show(io::IO, x::T) where {T <: RorC}
     end
 end
 
+function Base.show(io::IO, table::AbstractColumns; max_cols = 20)
+    nrows = try
+        string(length(Tables.getcolumn(table, 1)))
+    catch e
+        e isa MethodError || rethrow()
```

Comment:
trying to catch a `MethodError` here if `length` isn't supported, but I wonder if there's a better way to do this

Reply:
maybe use `IteratorSize` trait? I am not sure if this would be better.

Reply:
We can assume columns have length; they have to be 1-indexable.

Reply:
We also have `Tables.rowcount` that already does this

Reply:
ah great, thanks!